code/modules/awaymissions/corpse.dm

```diff
@@ -43,8 +43,8 @@
 		return
 	if(isobserver(user))
 		var/mob/dead/observer/O = user
-		if(!O.can_reenter_round)
-			to_chat(user, "<span class='warning'>You are unable to reenter the round.</span>")
+		if(O.reenter_round_timeout > world.realtime)
+			to_chat(user, "<span class='warning'>You are unable to reenter the round yet. Your ghost role blacklist will expire in [round((O.reenter_round_timeout - world.realtime)/600)] minutes.</span>")
```

Comment:
If this is more than 4 hours (or if we have a config option that allows changing the total round time, whatever that is multiplied by 2), then the timer shouldn't be displayed. Also displaytimetext is a thing